tests/unit/test_arithmetic_expression.cpp

```diff
@@ -1411,3 +1411,20 @@ TEST_F(ArithmeticTest, ParamTest) {
 	for(int i = 0; i < 3; i++) SIArray_Append(&array, SI_LongVal(i));
 	ASSERT_EQ(0, SIValue_Compare(array, arExp->operand.constant, NULL));
 }
+
+TEST_F(ArithmeticTest, CoalesceTest) {
+	const char *query;
+	AR_ExpNode *arExp;
+
+	query = "RETURN coalesce(1)";
+	arExp = _exp_from_query(query);
+	ASSERT_EQ(AR_EXP_OPERAND, arExp->type);
+	ASSERT_EQ(AR_EXP_CONSTANT, arExp->operand.type);
+	ASSERT_EQ(0, SIValue_Compare(SI_LongVal(1), arExp->operand.constant, NULL));
+
+	query = "RETURN coalesce(null, 1)";
+	arExp = _exp_from_query(query);
+	ASSERT_EQ(AR_EXP_OPERAND, arExp->type);
+	ASSERT_EQ(AR_EXP_CONSTANT, arExp->operand.type);
+	ASSERT_EQ(0, SIValue_Compare(SI_LongVal(1), arExp->operand.constant, NULL));
```

Comment:
Same comment as above, worth mentioning this, otherwise this is a bit confusing.
I would add a flow test which checks against variables.
return COALESCE(n.a, n.b, n.c) where each time only one is not NULL, and another test which returns the default.

Reply:
added comments

Reply:
added comments
added flow test